burrito_link/dllmain.c

```diff
@@ -396,3 +396,55 @@ uintptr_t mod_release() {
 extern __declspec(dllexport) void* get_release_addr() {
     return mod_release;
 }
+
+
+////////////////////////////////////////////////////////////////////////////////
+// Raidcore Nexus
+//
+// GetAddonDef()
+// 
+//
+//
+// These function is present to allow nexus to recognize this dll as a addon.
+// These function is the only function that is required all others are optional.
+////////////////////////////////////////////////////////////////////////////////
+struct AddonDefinition {
+	/* required */
+	signed int      Signature;      /* Raidcore Addon ID, set to random unqiue negative integer if not on Raidcore */
+	signed int      APIVersion;     /* Determines which AddonAPI struct revision the Loader will pass, use the NEXUS_API_VERSION define from Nexus.h */
+	const char*     Name;           /* Name of the addon as shown in the library */
+	struct AddonVersion {
+		signed short      Major;
+		signed short      Minor;
+		signed short      Build;
+		signed short      Revision;
+	} Version;
+	const char*     Author;         /* Author of the addon */
+	const char*     Description;    /* Short description */
+	void*           Load;           /* Pointer to Load Function of the addon */
+	void*           Unload;         /* Pointer to Unload Function of the addon. Not required if EAddonFlags::DisableHotloading is set. */
+	signed int      Flags;          /* Information about the addon */
+
+	/* update fallback */
+	signed int      Provider;       /* What platform is the the addon hosted on */
+	const char*     UpdateLink;     /* Link to the update resource */
+
+} AddonDef;
+
+extern __declspec(dllexport) struct AddonDefinition* GetAddonDef()
+{
+	AddonDef.Signature = -217;
+	AddonDef.APIVersion = 6; // taken from Nexus.h
+	AddonDef.Name = "Burrito Link";
+	AddonDef.Version.Major = 1;
+	AddonDef.Version.Minor = 4;
+	AddonDef.Version.Build = 0;
+	AddonDef.Version.Revision = 1;
```

Comment:
I think these values should probably be dynamically populated, almost certainly  via preprocesor values passed in via cmake/make. Semver is always an annoying methodology to follow because it usually requires either manual intervention or a persisted git-adjacent datastore. Maybe we should just transition to a date-based system like many other programs use in order to avoid having to manually track this. I'll need to think about that some more. Feel free to propose suggestions.

Reply:
I think this depends on whether we want to work towards the advanced PR (including updates).
For arcdps we always report the version as `arc_exports.out_build = "1.0";` we could do the same here.

But in general it would be better to show the real version, alone for reporting errors and such.
I will see if can come up with a good suggestion.

Reply:
Maybe like this.

```
string(REPLACE "." ";" BURRITO_VERSION_LIST ${BURRITO_VERSION})
list(LENGTH BURRITO_VERSION_LIST BURRITO_VERSION_LIST_LENGTH)
if(BURRITO_VERSION_LIST_LENGTH EQUAL 4)
    list(GET BURRITO_VERSION_LIST0 BURRITO_VERSION_MAJOR)
    list(GET BURRITO_VERSION_LIST1 BURRITO_VERSION_MINOR)
    list(GET BURRITO_VERSION_LIST2 BURRITO_VERSION_BUILD)
    list(GET BURRITO_VERSION_LIST2 BURRITO_VERSION_REVISION)
else()
    message( FATAL_ERROR "Version format mismatch." )
endif()

add_compile_definitions(BURRITO_VERSION_MAJOR=\"${BURRITO_VERSION_MAJOR}\")
add_compile_definitions(BURRITO_VERSION_MINOR=\"${BURRITO_VERSION_MINOR}\")
add_compile_definitions(BURRITO_VERSION_BUILD=\"${BURRITO_VERSION_BUILD}\")
add_compile_definitions(BURRITO_VERSION_REVISION=\"${BURRITO_VERSION_REVISION}\")
```

Reply:
I suggest that we proceed as follows.

1. We merge this PR as is, with hardcoded values for the version.
    At the moment none of the executables or libraries we build is aware of its version, and there is no need to change this with this PR.
2. We publish a new release with the current version scheme, that would be `alpha-1.5` I think.
   There are enough changes/fixes in the `main` branch to have a new release anyway and so we have a working dll in the release-bundle. This is what matters most for our users.
3. I will open another PR where we can discuss the version scheme (I don' see whats the problem with semantic versioning and git-tags) and implement "version awareness" (don't know how else to call this) in every binary we build.
4. We can revisit the #327 to make automatic updates work if we want.